src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        Extensions.[|ExtensionM1|](new object(), 1);
+    }
+}}";
+            var fix =
+@"
+namespace N {
+static class Extensions
+{
+    public static void ExtensionM1(this object o, int v)
+    {
+    }
+}
+class C1
+{
+    void M1()
+    {
+        Extensions.ExtensionM1(new object(), 1);
+    }
+}}";
+            await TestInRegularAndScriptAsync(code, fix);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationOverride()
+        {
+            var code = @"
+class Base
+{
+    protected virtual void M1() { }
+}
+class C1 : Base
+{
+    protected override void M1() { }
+    void M2()
+    {
+        [|M1|](1);
+    }
+}";
+            var fix_DeclarationOnly = @"
+class Base
+{
+    protected virtual void M1() { }
+}
+class C1 : Base
+{
+    protected override void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            var fix_All = @"
+class Base
+{
+    protected virtual void M1(int v) { }
+}
+class C1 : Base
+{
+    protected override void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix_DeclarationOnly, index: 0);
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationExplicitInterface()
+        {
+            var code = @"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+    void I1.M1() { }
+    void M2()
+    {
+        ((I1)this).[|M1|](1);
+    }
+}";
+            var fix_DeclarationOnly = @"
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    void I1.M1() { }
+    void M2()
+    {
+        ((I1)this).M1(1);
+    }
+}";
+            var fix_All = @"
+interface I1
+{
+    void M1(int v);
+}
+class C1 : I1
+{
+    void I1.M1(int v) { }
+    void M2()
+    {
+        ((I1)this).M1(1);
+    }
+}";
+            await TestInRegularAndScriptAsync(code, fix_DeclarationOnly, index: 0);
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocationImplicitInterface()
+        {
+            var code =
+@"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+    public void M1() { }
+    void M2()
+    {
+        [|M1|](1);
+    }
+}";
+            var fix_DeclarationOnly = @"
+interface I1
+{
+    void M1();
+}
+class C1 : I1
+{
+    public void M1(int v) { }
+    void M2()
+    {
+        M1(1);
+    }
+}";
+            var fix_All = @"
+interface I1
+{
+    void M1(int v);
```

Comment:
if C1 implements 2 interface and if both of them has method name M1, will this update both interfaces?

Reply:
Because this features relies on *find all references*, I assume both will be fixed. I can a test if you like.

Reply:
test would be nice if that is not too much trouble.

Reply:
Done in a98dabcdff0d261b190d50bd3a47399d3c2bae7d.